lib/generator.go

```diff
@@ -75,58 +83,106 @@ func GenerateCCRs(config *Config, requests chan *request) (err error) {
 	// ccrsPerDay = 1440m / 30m = 40 CCR a day
 	// ccrsTotal = ccrsPerDay * 30d = 1200 per Node
 	if config.DaysBack > 0 {
-		ccrsPerDay = 1440 / config.Interval
-		ccrsTotal = ccrsPerDay * config.DaysBack
+		ccrsPerDay = 1440 / int64(config.Interval)
+		ccrsTotal = ccrsPerDay * int64(config.DaysBack)
 	}
 
 	if config.RunChefClient {
 		chefClient = getAPIClient(config.ClientName, config.ClientKey, config.ChefServerURL)
 	}
 
 	log.WithFields(log.Fields{
-		"nodes":       config.NumNodes,
-		"days_back":   config.DaysBack,
-		"random_data": config.RandomData,
+		"nodes":        config.NumNodes,
+		"days_back":    config.DaysBack,
+		"ccr_per_node": ccrsTotal,
+		"total_ccrs":   ccrsTotal * int64(config.NumNodes),
+		"random_data":  config.RandomData,
+		"goroutines":   chanSize,
 	}).Info("Generating chef-client runs")
 
 	rand.Seed(time.Now().UTC().UnixNano())
 
+	// Lets try to use a smaller number of goroutines
+	if config.NumNodes > chanSize {
+		// If the number of nodes is bigger than the channel
+		// size, lets calculate how many loops we need to run
+		loops = config.NumNodes / chanSize
```

Comment:
This might not be a whole number. Which the number would be truncated.

Reply:
For now we can just pick numbers for NumNodes and chanSize that divisible.

Reply:
I see on line 122 you are dealing with the truncation of the division here. I think you would need to do a math.Ceil for that to be useful.